File: src/cowbird/initdb.clj
(ns cowbird.initdb
  (:require [clojure.tools.cli :refer [parse-opts]]
            [aleph.tcp :as tcp]
            [manifold.stream :as s]
            [clojure.java.io :as io]))

(def initdb-cli-options
  [["-D" "--pgdata PGDATADIR" nil ]
   [nil "--noclean" nil]
   [nil "--nosync" nil]])

(defn initdb [args]
  (let [args (parse-opts args initdb-cli-options)
        sample-conf
        (io/file (io/resource "postgres/src/backend/utils/misc/postgresql.conf.sample"))
        conf-file
        (io/file (str (get-in args [:options :pgdata]) "/postgresql.conf"))]
    (println args)
    (io/make-parents conf-file)
    (io/copy sample-conf conf-file)
    (System/exit 0)))

File: lib/cowbird/initdb.ex
defmodule Cowbird.Initdb do
  def main(args) do
    IO.puts("Running Initdb")
    IO.inspect(args)
    {opts,_,_}= OptionParser.parse(args,
      switches:
      [directory: :string,
       noclean: :boolean,
       nosync: :boolean],
      aliases: [D: :directory])
    File.mkdir(opts[:directory])
  end
end 
